iceoryx_utils/include/iceoryx_utils/internal/concurrent/fifo.inl

```diff
@@ -51,17 +54,23 @@ inline bool FiFo<ValueType, Capacity>::empty() const
 template <class ValueType, uint32_t Capacity>
 inline cxx::optional<ValueType> FiFo<ValueType, Capacity>::pop()
 {
-    if (empty())
+    bool isEmpty = (m_read_pos.load(std::memory_order_relaxed) ==
+                    // we are not allowed to use the empty method since we have to sync with
+                    // the producer pop - this is done here
+                    m_write_pos.load(std::memory_order_acquire));
+    if (isEmpty)
     {
         return cxx::nullopt_t();
     }
     else
     {
-        ValueType out = m_data[m_read_pos.load(std::memory_order_acquire) % Capacity];
+        auto currentReadPos = m_read_pos.load(std::memory_order_relaxed);
```

Comment:
why do you read the read position here again? you already did it for the empty check, just reuse that one

Reply:
This was on purpose - I wanted to see if you can spot it ;)